src/Angular.js

```diff
@@ -767,6 +767,36 @@ function equals(o1, o2) {
   return false;
 }
 
+/**
+ * @ngdoc directive
+ * @name ng.directive:ngJq
+ *
+ * @element html
+ * @param {string} (optional) the name of the library available under `window` to be used for angular.element
+ *
+ * @description
+ *
+ * Use this directive to force the angular.element library.  This should be used to force
+ * either jqLite by leaving ng-jq blank or setting the name of the jquery variable under window
+ * (eg. jQuery).
+ *
+ * This directive is global for the whole of the angular library, hence the directive can only
+ * be added to the top-most HTML element.
+ *
+ */
+var jq = function () {
+  if (isDefined(jq.name_)) return jq.name_;
+
+  var el = document.querySelector('[ng-jq]') || document.querySelector('[data-ng-jq]');
```

Comment:
@lgalfaso I'm going on @mzgol suggestions on how to implement this, which was based on ng-csp since angular.element is a global variable, unlike ng-app which you can have multiple in a page.

As for the branch itself, this was the latest that I could get from github, fetching the latest from upstream and making sure that my master has the same last commit than the angular one.  If there's any branch 'weirdness', I can't comment why that would be.